spec/models/ems_event_spec.rb

```diff
@@ -519,4 +519,25 @@
       end
     end
   end
+
+  context 'Physical Storage Events' do
+    let(:ems)   { FactoryBot.create(:ems_autosde) }
+    let(:physical_storage) { FactoryBot.create(:physical_storage_autosde, :ems_ref => "ems1", :ext_management_system => ems) }
```

Comment:
Best to stick to generic classes rather than provider specific ones in core:
```suggestion
    let(:ems)              { FactoryBot.create(:ems_storage) }
    let(:physical_storage) { FactoryBot.create(:physical_storage, :name => "my-storage", :ext_management_system => ems) }
```